src/Interpreters/SystemLogStorage.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <Interpreters/StorageID.h>
+#include <Interpreters/Context_fwd.h>
+
+namespace DB
+{
+
+template <typename LogElement>
+class SystemLogStorage: private boost::noncopyable, protected WithContext
```

Comment:
What's the difference with `SystemLog` ?

Reply:
Yes, @sirvickr borrowed much code from `SystemLog`. I think he will explain in details why he did it. And how it finally work, for example, why `SystemLogStorage` does not have `update` method. In my opinion probably some common base class should be used for both  `SystemLog` and `SystemLogStorage` or some other OOP approaches in order to avoid code duplication.

Reply:
I've already implemented this refactoring (SystemLog and BackupsStorage both use SystemLogStorage class) and it seems to work locally so if this PR is accepted in a current form, I can create a follow-up PR with the refactoring.